File: app/src/main/java/waldrapps/plannit/activities/NewContactActivity.java
package waldrapps.plannit.activities;

import android.os.Bundle;
import android.support.v4.app.FragmentActivity;
import android.view.Menu;
import android.view.MenuItem;

import waldrapps.plannit.R;
import waldrapps.plannit.fragments.NewContactFragment;

public class NewContactActivity extends BaseFragmentActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
//        setTheme(R.style.CustomTheme1);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_new_contact);
        //Start new contact fragment
        if (savedInstanceState == null) {
            //Get uuid
            String uuid = getIntent().getStringExtra("uuid");
            //Pass uuid to fragment
            Bundle bundle = new Bundle();
            bundle.putString("uuid", uuid);
            NewContactFragment fragment = NewContactFragment.newInstance();
            fragment.setArguments(bundle);
            getSupportFragmentManager().beginTransaction().add(R.id.fragment_container, fragment).commit();
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_new_contact, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/waldrapps/plannit/activities/NewContactActivity.kt
package waldrapps.plannit.activities

import android.os.Bundle
import kotlinx.android.synthetic.main.activity_new_contact.*
import waldrapps.plannit.R
import waldrapps.plannit.fragments.NameDialogFragment
import waldrapps.plannit.fragments.NewContactFragment

class NewContactActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_new_contact)

        setSupportActionBar(toolbar)
        //Set back button
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setDisplayShowHomeEnabled(true)
        //Get name initially
        NameDialogFragment.newInstance(null).show(supportFragmentManager, "NameDialogFragment")
        //Allow user to edit name by clicking on title
        toolbar.setOnClickListener { NameDialogFragment.newInstance(toolbar.title.toString()).show(supportFragmentManager, "NameDialogFragment") }

        //Start new contact fragment
        if (savedInstanceState == null) {
            //Pass uuid to fragment
            val bundle = Bundle()
            bundle.putString(getString(R.string.uuid), intent.getStringExtra(getString(R.string.uuid)))
            //Start fragment
            val fragment = NewContactFragment.newInstance()
            fragment.arguments = bundle
            supportFragmentManager.beginTransaction().add(R.id.fragment_container, fragment).commit()
        }
    }
}